synapse/handlers/pagination.py

```diff
@@ -348,6 +372,10 @@ async def _purge_history(
             room_id: The room to purge from
             token: topological token to delete events before
             delete_local_events: True to delete local events as well as remote ones
+            update_rooms_to_delete_table: True if we don't want to update/persist this
+                purge history action to the DB to be restorable. Used with the retention
+                functionality since we don't need to explicitly restore those, they
+                will be relaunch by the retention logic.
```

Comment:
If we don't care about retrying the purge history tasks, why do store them in the database?

Reply:
We do care about retrying manual purge history tasks (launched through the purge history API), but not the ones launched by the retention logic: those will be relaunched anyway in the loop dealing with retention.